mslib/mscolab/server.py

```diff
@@ -351,6 +352,39 @@ def get_user():
     return json.dumps({'user': {'id': g.user.id, 'username': g.user.username}})
 
 
+@APP.route('/upload_profile_image', methods=["POST"])
+@verify_user
+def upload_profile_image():
+    user_id = request.form['user_id']
+    file = request.files['image']
+    if file:
+        if file.mimetype.startswith('image/'):
+            if file.content_length > mscolab_settings.MAX_IMAGE_SIZE:
+                return jsonify({'message': 'File too large'}), 413
+            success, message = fm.save_user_profile_image(user_id, file)
+            if success:
+                return jsonify({'message': message}), 200
+            else:
+                return jsonify({'message': message}), 400
+        else:
+            return jsonify({'message': 'Invalid file type'}), 400
+    else:
+        return jsonify({'message': 'No file provided or invalid file type'}), 400
```

Comment:
A lot of the nesting in this function could be avoided by inverting the conditions and returning early. See https://softwareengineering.stackexchange.com/questions/18454/should-i-return-from-a-function-early-or-use-an-if-statement for an example of what I mean (specifically the second answer). This would make the function much more readable.